eos/b-decays/signal-pdfs.cc

```diff
@@ -0,0 +1,453 @@
+/* vim: set sw=4 sts=4 et tw=150 foldmethod=marker : */
+
+/*
+ * Copyright (c) 2023 Danny van Dyk
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/signal-pdf-impl.hh>
+#include <eos/b-decays/b-to-d-l-x-nu.hh>
+#include <eos/b-decays/b-to-gamma-l-nu.hh>
+#include <eos/b-decays/b-to-psd-l-nu.hh>
+#include <eos/b-decays/b-to-pi-l-x-nu.hh>
+#include <eos/b-decays/b-to-pi-pi-l-nu.hh>
+#include <eos/b-decays/b-to-vec-l-nu.hh>
+#include <eos/b-decays/b-to-3l-nu.hh>
+#include <eos/b-decays/bs-to-kstar-l-nu.hh>
+#include <eos/b-decays/lambdab-to-lambdac-l-nu.hh>
+#include <eos/b-decays/lambdab-to-lambdac2625-l-nu.hh>
+#include <eos/utils/concrete-signal-pdf.hh>
+
+namespace eos
+{
+    // Leptonic and photoleptonic B decays
+    // {{{
+    SignalPDFGroup
+    make_b_to_leptons_pdf_group()
+    {
+        auto imp = new Implementation<SignalPDFGroup>(
+            R"(Signal PDFs in leptonic and photoleptonic $B$ decays)",
+            R"()",
+            {
+                // B -> gamma l nu
+                make_signal_pdf("B->gammalnu::d^2Gamma/dEgamma/dcos(theta_l)",
+                    Options{ },
+                    &BToGammaLeptonNeutrino::fully_differential_decay_width,
+                    std::make_tuple(
+                        KinematicRange{ "Egamma", 0.1, 2.64, BToGammaLeptonNeutrino::kinematics_description_Egamma },
+                        KinematicRange{ "cos(theta_l)", -1.0, +1.0, BToGammaLeptonNeutrino::kinematics_description_c_theta_l}
+                    ),
+                    &BToGammaLeptonNeutrino::integrated_branching_ratio,
+                    std::make_tuple(
+                        "E_gamma_min"
```

Comment:
Can we use the same convention `E_gamma` or `Egamma` throughout EOS?

Reply:
Yes. That should be done more systematically though. I have opened #750 and will resolve this conversation.